modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -66,12 +67,15 @@ object Run extends ScalaCommand[RunOptions] {
       build: Build.Successful,
       allowTerminate: Boolean
     ): Either[BuildException, (Process, CompletableFuture[_])] = either {
+      val potentialMainClasses = build.foundMainClasses()
+      value(options.mainClass.maybePrintMainClasses(potentialMainClasses))
```

Comment:
Ideally, if `allowTerminate` is false (in watch mode basically), we'd rather not have `maybePrintMainClasses` call `sys.exit` here. In that case, we should print the main class list, and that's it. Maybe we should change the return type, so that when listing main classes, this can return a `Right(…)`, without providing a `Process` and a `CompletableFuture`.

Reply:
That way, we'd print the main class list upon every source change.

Reply:
Hey, this is a great idea, didn't think of it.
It should now work with `--watch` mode after my last commit.